docs/transactions.rst

```diff
@@ -0,0 +1,187 @@
+Sending Transactions
+====================
+
+.. note::
+
+  Prefer to view this code in a Jupyter Notebook? View the repo `here <https://github.com/wolovim/ethereum-notebooks/blob/master/Sending%20Transactions.ipynb>`_.
+
+There are two methods for sending transactions using web3.py: :meth:`~web3.eth.Eth.send_transaction` and :meth:`~web3.eth.Eth.send_raw_transaction`. A brief guide:
+
+#. Want to sign a transaction offline or send pre-signed transactions?
+
+   * use :meth:`sign_transaction <eth_account.account.Account.sign_transaction>` + ``send_raw_transaction``
+
+#. Are you primarily using the same account for all transactions and would you prefer to save a few lines of code?
+
+   * configure :meth:`~web3.middleware.construct_sign_and_send_raw_middleware`, then
+   * use ``send_transaction``
+
+#. Otherwise:
+
+   * load account via eth-account (``w3.eth.account.from_key(pk)``), then
+   * use ``send_transaction``
+
+Interacting with or deploying a contract?
+
+* Option 1: :meth:`~web3.contract.ContractFunction.transact` uses ``send_transaction`` under the hood
+* Option 2: :meth:`~web3.contract.ContractFunction.build_transaction` + ``sign_transaction`` + ``send_raw_transaction``
+
+An example for each can be found below.
```

Comment:
It looks like this is a mix of links and `code` references. Is that on purpose? I think it would be nice if they all linked to their documentation, but feel free to take it or leave it!

Reply:
thought process applied here was to link the method only the first time it appears, as not to distract from the content. maybe there's an argument for it being more distracting though, if that's not obvious. i don't mind updating all to links, if that's preferable.

Reply:
Ah, gotcha. That wasn't obvious to me until you pointed it out, but makes sense. I'll defer to your writing expertise! :)

Reply:
opted to just link everything. the red formatting really harshes the vibe, man. ✌️